models/JSONPrettyPrint.cfc

```diff
@@ -4,147 +4,59 @@
 * www.coldbox.org | www.ortussolutions.com
 ********************************************************************************
 * @author Brad Wood, Luis Majano
-* 
+*
 */
 component accessors="true" singleton alias='JSONPrettyPrint' {
 
-	function init() {
-		variables.os = createObject( "java", "java.lang.System" ).getProperty( "os.name" ).toLowerCase();
-		return this;
-	}
-		
-	// OS detector
-	private boolean function isWindows(){ return variables.os.contains( "win" ); }
+    property CFMLPrinter;
+    property JSONPrinter;
+    property DefaultPrinter;
 
-	/**
-	 * Pretty JSON
-	 * @json A string containing JSON, or a complex value that can be serialized to JSON
-	 * @indent String to use for indenting lines.  Defaults to four spaces.
-	 * @lineEnding String to use for line endings.  Defaults to CRLF on Windows and LF on *nix
-	 * @spaceAfterColon Add space after each colon like "value": true instead of"value":true 
- 	 **/
-	public function formatJson( any json, string indent='    ', lineEnding, boolean spaceAfterColon=false ) {
-		
-		// Default line ending based on OS
-		if( isNull( arguments.lineEnding ) ) {
-			if( isWindows() ) {
-				arguments.lineEnding = chr( 13 ) & chr( 10 );
-			} else {
-				arguments.lineEnding = chr( 10 );				
-			}
-		}
-		
-		// Overload this method to accept a struct or array
-		if( !isSimpleValue( arguments.json ) ) {
-			arguments.json = serializeJSON( arguments.json );
-		}
-		
-		arguments.json = arguments.json.replace( '	', '', 'all' );
-		arguments.json = arguments.json.replace( chr( 10 ), '', 'all' );
-		arguments.json = arguments.json.replace( chr( 13 ), '', 'all' );
-		
-		var retval = createObject( 'java', 'java.lang.StringBuilder' ).init( '' );
-		var str = json;
-		var strLen = str.len();
-	    var pos = 0;
-	    var strLen = str.length();
-		var indentStr = arguments.indent;
-	    var newLine = arguments.lineEnding;
-		var char = '';
-		var nextChar = '';
-		var inQuote = false;
-		var isEscaped = false;
-		var itemsInCollection = [];
+    function init( string defaultPrinter = '' ) {
```

Comment:
I personally avoid constructor args.  They're a pain to pass in via Wirebox anyway since an object factory is creating our CFC instances. If a user wants to configure the state of this CFC, give them a method to do so.  I would be a little wary of all the options.  This has always been a pretty simple library but it feels like it's turning into a swiss army knife with all the ways to use it.

Reply:
Hey @bdw429s, I agree that I have complicated things. I am happy to back this all out. I understand you want a simple library.

Reply:
Let's not throw the baby out with the bathwater.  I like the sorting enhancements, it's just all the extra work to accommodate Adobe that seems like a lot.  I'm afraid we'll need a flowchart infographic for people to figure how which option they need to use for their engine/version/need :)  

It's ok to let people override the printer to be used, but can we move it from a constructor arg to a setter?  So instead of 
```js
jpp = new JSONPrettyPrint( defaultPrinter ='foobar' );
```
it's
```js
jpp = new JSONPrettyPrint();
jpp.setDefaultPrinter( 'foobar' );
```
That will leave it configurable but doesn't get in the way of the construction of the component and works with or without WireBox.